assignment-client/src/avatars/AvatarMixer.cpp

```diff
@@ -277,76 +274,70 @@ void AvatarMixer::broadcastAvatarData() {
-                        otherNode->getLastSequenceNumberForPacketType(PacketTypeAvatarData));
-
-                    QByteArray avatarByteArray;
-                    avatarByteArray.append(otherNode->getUUID().toRfc4122());
-                    avatarByteArray.append(otherAvatar.toByteArray());
-                    
-                    if (avatarByteArray.size() + mixedAvatarByteArray.size() > MAX_PACKET_SIZE) {
-                        nodeList->writeDatagram(mixedAvatarByteArray, node);
-
-                        numAvatarDataBytes += mixedAvatarByteArray.size();
-                            
-                        // reset the packet
-                        mixedAvatarByteArray.resize(numPacketHeaderBytes);
-                    }
-                        
-                    // copy the avatar into the mixedAvatarByteArray packet
-                    mixedAvatarByteArray.append(avatarByteArray);
-                        
+                    nodeData->setLastBroadcastSequenceNumber(otherNode->getUUID(),
+                        otherNode->getLastSequenceNumberForPacketType(PacketType::AvatarData));
+
+                    // start a new segment in the PacketList for this avatar
+                    avatarPacketList.startSegment();
+
+                    numAvatarDataBytes += avatarPacketList.write(otherNode->getUUID().toRfc4122());
+                    numAvatarDataBytes += avatarPacketList.write(otherAvatar.toByteArray());
+
+                    avatarPacketList.endSegment();
+
                     // if the receiving avatar has just connected make sure we send out the mesh and billboard
                     // for this avatar (assuming they exist)
                     bool forceSend = !nodeData->checkAndSetHasReceivedFirstPackets();
-                        
+
                     // we will also force a send of billboard or identity packet
                     // if either has changed in the last frame
-                        
+
                     if (otherNodeData->getBillboardChangeTimestamp() > 0
                         && (forceSend
                             || otherNodeData->getBillboardChangeTimestamp() > _lastFrameTimestamp
                             || randFloat() < BILLBOARD_AND_IDENTITY_SEND_PROBABILITY)) {
-                        QByteArray billboardPacket = nodeList->byteArrayWithPopulatedHeader(PacketTypeAvatarBillboard);
-                        billboardPacket.append(otherNode->getUUID().toRfc4122());
-                        billboardPacket.append(otherNodeData->getAvatar().getBillboard());
 
-                        nodeList->writeDatagram(billboardPacket, node);
-                            
+                        auto billboardPacket = NLPacket::create(PacketType::AvatarBillboard);
+                        billboardPacket->write(otherNode->getUUID().toRfc4122());
+                        billboardPacket->write(otherNodeData->getAvatar().getBillboard());
```

Comment:
can make this a bit smarter by grabbing those and figuring out what size they will be